File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/Results.java
package work.socialhub.kmastodon.entity;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * @author hecateball
 */
public class Results implements Serializable {

    @SerializedName("accounts")
    private Account[] accounts;
    @SerializedName("statuses")
    private Status[] statuses;
    @SerializedName("hashtags")
    private Tag[] hashtags;

    public Account[] getAccounts() {
        return accounts;
    }

    public void setAccounts(Account[] accounts) {
        this.accounts = accounts;
    }

    public Status[] getStatuses() {
        return statuses;
    }

    public void setStatuses(Status[] statuses) {
        this.statuses = statuses;
    }

    public Tag[] getHashtags() {
        return hashtags;
    }

    public void setHashtags(Tag[] hashtags) {
        this.hashtags = hashtags;
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/Results.kt
package work.socialhub.kmastodon.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class Results {

    @SerialName("accounts")
    var accounts: Array<Account>? = null

    @SerialName("statuses")
    var statuses: Array<Status>? = null

    @SerialName("hashtags")
    var hashtags: Array<Tag>? = null
}
